### ISSUE TITLE
Site Kit installed GTM only active while logged into WP admin

### ISSUE THREAD

**Issue by frankblob** [google/site-kit-wp#244]
## Bug Description

I installed Google Tag Manager (GTM) as part of Site Kit in Wordpress. 

My GTM account has a single container, with a single tag (FB pixel) and a single trigger (all pages). GTM works as intended, when I am logged into WP admin. FB Pixel helper reports a FB Pixel is present. And I receive test data in Facebook for the pixel, as expected.

As soon as I log out of WP admin, the GTM container code changes and the FB Pixel helper reports that no Pixels present. And the FB pixel changes from active.

## Additional Context

Server Environment		
Operating System:	Linux							
Software:	Apache/2.4.10 (Debian)							
MySQL version:	Percona XtraDB Cluster (GPL), Release rel76.3, Revision 0a59e2f, WSREP version 25.16, wsrep_25.16 v5.6.30-76.3							
PHP Version:	7.2.8							
PHP Max Input Vars:	3000							
PHP Max Post Size:	100M							

WordPress Version: 5.2.2
Memory limit: 256M
Site Kit version 1.0.0-beta.1.0.1	

Computer:
 - Browser Chrome Version 75.0.3770.100 (Official Build) (64-bit)
 - Device: Laptop running Win10

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

## Changelog entry

*

**Comment by jamesozzie** [comment]
@frankblob Thanks for raising this issue. Can you inform me of the following so we can troubleshoot further:
1. Did you have any Tag Manager implemented before Site Kit (via a plugin or manual insertion)?
2. Can you confirm you see the source code as below on your website when you are logged in and also logged out of your dashboard:
`<!-- Google Tag Manager added by Site Kit -->...`

If you want to attempt disconnecting the Tag Manager module from Site Kit before temporarily deactivating other plugins and trying again we can see if that works.

**Comment by frankblob** [comment]
Hi @jamesozzie. 
    1. The site not have GA, GTM or a FB pixel installed previously. Neither manually nor via any plugins. 
    2. The code is only present, when logged in. It seems it is activated as part of the admin. 
        Logged in/admin: In regular (as you listed it) and a <noscript> version, too.
        Logged out: `<meta name="generator" content="Site Kit by Google 1.0.0-beta.1.0.1" />` is the only code present in regards to Site Kit. The regular and <noscript> is not there.

In plain English - the `<!-- Google Tag Manager added by Site Kit -->` is only in the source code, while logged in as admin. But the page holds the `<meta name="generator" content="Site Kit by Google 1.0.0-beta.1.0.1" />` in both states - logged in and logged out.

**Comment by frankblob** [comment]
Progress. After disabling plugins one-by-one, I found the culprit, as you suggested. Autoptimize. After flushing the cache and reenabling the Autoptimize plugin, GTM now works as intended - also, when logged out. Thanks.

**Comment by jamesozzie** [comment]
@frankblob Thanks for letting us know, very useful for others who will potentially face the same issue.
